common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -104,7 +104,10 @@ def target_branch
 
       sig { returns(T::Array[DependencyFile]) }
       def files
-        @files ||= T.let(fetch_files, T.nilable(T::Array[DependencyFile]))
+        @files ||= T.let(
+          fetch_files.each { |f| f.job_directory = directory },
```

Comment:
I can't believe I never realized `Array#each` returns the original array 😲.

Reply:
I had to hop into `irb` to check 😁.

At first I tried refactoring to use the `return... if defined?` pattern but the types were having me sprinkle `T.must` everywhere which was quite ugly. This seemed more elegant.